docs/source/developers/documentation.rst

```diff
@@ -87,6 +87,14 @@ These two steps are mandatory and must be executed in order.
    not sufficiently comprehensive. Portions of the Python API documentation
    will also not build without CUDA support having been built.
 
+.. note::
+
+   If you are working on the Python documentation and are building the documentation
+   with ``pyarrow`` build from source on macOS Monterey, Python section of the
+   documentation might not be included in the ``_build/html``. In this case run
+   ``python -m pip install ../python --quiet --no-dependencies python`` before
+   running the ``make html`` command.
```

Comment:
This is essentially installing the package in non-editable mode, right? Maybe mention that explicitly to clarify what the command is exactly doing

Also, what does the `--no-dependencies python` do?

Reply:
> This is essentially installing the package in non-editable mode, right? Maybe mention that explicitly to clarify what the command is exactly doing

Yes, correct. Will do.

> Also, what does the --no-dependencies python do?

Last `python` is a typo. The `--no-dependencies` is something I saw others use but didn't bother to double check as it was working locally, sorry about that. Running `pip install -h` lists

```--no-deps                   Don't install package dependencies.```

The dependencies will already be installed in our case so this doesn't really have an effect I suppose. Will remove.